### PR TITLE
code separation and optimization

### LINKED ISSUE #1530: Separate Internal and exposed class from SDK and optimize code
**Describe the Issue**
 Mark LocalChangeEntity as internal and Create respective public class
Removed SquashedLocalChangeClass as it will duplicate above public class

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by PallaviGanorkar**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1530 
**Description**
Clear and concise code change description. 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Code health

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: engine/src/main/java/com/google/android/fhir/db/impl/entities/LocalChangeEntity.kt [hunk 1]
```diff
@@ -51,22 +52,12 @@ import androidx.room.PrimaryKey
  * ] For resource that is fully synced with server this table should not have any rows.
  */
 @Entity(indices = [Index(value = ["resourceType", "resourceId"])])
-data class LocalChangeEntity(
+internal data class LocalChangeEntity(
```
Comment (jingtang10) [inline engine/src/main/java/com/google/android/fhir/db/impl/entities/LocalChangeEntity.kt:55]: this is great

File: engine/src/main/java/com/google/android/fhir/db/LocalChangeType.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.db
+
+/** Enum specifies type of local change */
+enum class LocalChangeType(val value: Int) {
```
Comment (jingtang10) [inline engine/src/main/java/com/google/android/fhir/db/LocalChangeType.kt:20]: i think this type enum class can be in the same fiile as the local change class that is public. it's easier for people to read and find the type for that class.

but also - i think the type for local change entity, which is internal, should be restored. and you will need to  add a mapping between this type, and the internal type.

File: engine/src/main/java/com/google/android/fhir/LocalChange.kt [hunk 3]
```diff
@@ -28,10 +29,14 @@ data class LocalChange(
   /** last updated timestamp */
```
Comment (jingtang10) [inline engine/src/main/java/com/google/android/fhir/LocalChange.kt:29]: last updated locally? or on server? more details on what this is

File: engine/src/main/java/com/google/android/fhir/LocalChange.kt [hunk 4]
```diff
@@ -28,10 +29,14 @@ data class LocalChange(
...
   val timestamp: String = "",
   /** Type of local change like insert, delete, etc */
-  // TODO extract Type enum from LocalChangeEntity and make LocalChangeEntity as internal class
-  val type: LocalChangeEntity.Type,
+  val type: LocalChangeType,
   /** json string with local changes */
   val payload: String,
   /** last udated vesrion for resource */
-  val versionId: String? = null
+  val versionId: String? = null,
```
Comment (jingtang10) [inline engine/src/main/java/com/google/android/fhir/LocalChange.kt:36]: this is the id of the version of the resource that this local change is based off of.

Comment (jingtang10) [inline engine/src/main/java/com/google/android/fhir/LocalChange.kt:36]: move this to after resource id

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: engine/src/main/java/com/google/android/fhir/LocalChange.kt (hunk 4 above)
Comment (PallaviGanorkar) [reply engine/src/main/java/com/google/android/fhir/LocalChange.kt:36]: Done

Comment (PallaviGanorkar) [reply engine/src/main/java/com/google/android/fhir/LocalChange.kt:36]: Done

File: engine/src/main/java/com/google/android/fhir/LocalChange.kt (hunk 3 above)
Comment (PallaviGanorkar) [reply engine/src/main/java/com/google/android/fhir/LocalChange.kt:29]: Done

File: engine/src/main/java/com/google/android/fhir/db/LocalChangeType.kt (hunk 2 above)
Comment (PallaviGanorkar) [reply engine/src/main/java/com/google/android/fhir/db/LocalChangeType.kt]: Done

File: engine/src/main/java/com/google/android/fhir/db/impl/DbTypeConverters.kt [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 Google LLC
+ * Copyright 2022 Google LLC
```
Comment (jingtang10) [inline engine/src/main/java/com/google/android/fhir/db/impl/DbTypeConverters.kt:2]: revert this

**Review by jingtang10 — APPROVED** [review]

File: engine/src/main/java/com/google/android/fhir/db/impl/DbTypeConverters.kt (hunk 5 above)
Comment (PallaviGanorkar) [reply engine/src/main/java/com/google/android/fhir/db/impl/DbTypeConverters.kt]: done

**Pull request merged**